### PR TITLE
Java cleanup

### PR DESCRIPTION
**Pull request by jsoref**
This is for discussion.

Notes:
* Most commits were created by IntelliJ (the descriptions are mine because I lost the details)
* Individual commits are mostly self contained
* In general, most commits should not change program behavior at all

### Proposed changelog entries

* `Internal:` Internal Java code cleanup

### Submitter checklist

- [ ] JIRA issue is well described
- [ ] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### CODE REVIEW

File: core/src/main/java/hudson/Main.java [hunk 1]
```diff
@@ -152,9 +153,8 @@ public static int remotePost(String[] args) throws Exception {
                 // run the command
                 long start = System.currentTimeMillis();
 
-                List<String> cmd = new ArrayList<String>();
-                for( int i=1; i<args.length; i++ )
-                    cmd.add(args[i]);
+                List<String> cmd = new ArrayList<>();
+                cmd.addAll(Arrays.asList(args).subList(1, args.length));
```
Comment (jvz) [inline core/src/main/java/hudson/Main.java]: Neat! Reminds me of Scala a bit for some reason.

File: core/src/main/java/hudson/model/Computer.java [hunk 2]
```diff
@@ -199,7 +199,7 @@
      * @see Executor#resetWorkUnit(String)
      */
     private transient final List<TerminationRequest> terminatedBy = Collections.synchronizedList(new ArrayList
-            <TerminationRequest>());
+            <>());
```
Comment (jvz) [inline core/src/main/java/hudson/model/Computer.java]: Seems like this line could be collapsed now?

File: core/src/main/java/hudson/model/Fingerprint.java [hunk 3]
```diff
@@ -1199,13 +1199,13 @@ public int size() {
      * @return Sorted list of {@link FingerprintFacet}s 
      */
     public @Nonnull Collection<FingerprintFacet> getSortedFacets() {
-        List<FingerprintFacet> r = new ArrayList<FingerprintFacet>(getFacets());
-        Collections.sort(r,new Comparator<FingerprintFacet>() {
+        List<FingerprintFacet> r = new ArrayList<>(getFacets());
+        r.sort(new Comparator<FingerprintFacet>() {
```
Comment (jvz) [inline core/src/main/java/hudson/model/Fingerprint.java]: Seems like you can likely use `Comparator.comparingLong(FingerprintFacet::getTimestamp)` instead of the manually written one.

File: core/src/main/java/hudson/model/Queue.java [hunk 4]
```diff
@@ -73,6 +73,8 @@
 import jenkins.util.SystemProperties;
 import jenkins.util.Timer;
 import hudson.triggers.SafeTimerTask;
+
+import java.util.*;
```
Comment (jvz) [inline core/src/main/java/hudson/model/Queue.java]: Might not want to do that, especially with how Java adds new classes to this package all the time it seems.

File: core/src/main/java/hudson/model/TaskThread.java [hunk 5]
```diff
@@ -188,7 +188,7 @@ public static ListenerAndText forMemory(TaskAction context) {
 
             return new ListenerAndText(
                 new StreamTaskListener(log),
-                new AnnotatedLargeText<TaskAction>(log,Charset.defaultCharset(),false,context)
+                    new AnnotatedLargeText<>(log, Charset.defaultCharset(), false, context)
```
Comment (jvz) [inline core/src/main/java/hudson/model/TaskThread.java]: Not sure what happened with the formatting here?

**Review by jvz — APPROVED** [review]
Looks good overall. Have a couple suggestions.

File: core/src/main/java/hudson/model/Computer.java (hunk 2 above)
Comment (jsoref) [reply core/src/main/java/hudson/model/Computer.java]: I'm not really pretending to speak Java at this point. There are more commits in my queue:
https://github.com/jsoref/jenkins/tree/java-cleanup-full

But, these are (generally) things that IntelliJ is suggesting.

File: core/src/main/java/hudson/model/Queue.java (hunk 4 above)
Comment (jsoref) [reply core/src/main/java/hudson/model/Queue.java]: This is IntelliJ -- it has some threshold after which it decides
https://www.jetbrains.com/help/idea/creating-and-optimizing-imports.html:
`Class count to use import with ‘*’` (`5`)
`Names count to use static import with ‘*’` (`3`)

If people strenuously object, I can adjust, but, we're talking to a computer program...

File: core/src/main/java/hudson/model/TaskThread.java (hunk 5 above)
Comment (jsoref) [reply core/src/main/java/hudson/model/TaskThread.java]: My guess is the parser got confused (by the `(` & `,` on the previous line?) or has a different opinion about the indentation rules.

File: core/src/main/java/hudson/model/Fingerprint.java (hunk 3 above)
Comment (jsoref) [reply core/src/main/java/hudson/model/Fingerprint.java]: It looks like that happens [often in jenkins](https://github.com/jenkinsci/jenkins/search?q=%22new+Comparator%22&unscoped_q=%22new+Comparator%22), I might go back and add a commit later which does that cleanup: https://github.com/jenkinsci/jenkins/blob/22aa2e6e766074d11249893e3f35e0b99e20d3d0/test/src/test/java/jenkins/model/ExtensionTypeHierarchyTest.java#L55
https://github.com/jenkinsci/jenkins/blob/0795e89b308ec7fcbda097858d58763d8531be8c/core/src/main/java/hudson/views/MyViewsTabBar.java#L89
https://github.com/jenkinsci/jenkins/blob/d9aa1283e692f7fc1e19b8e360a971871ce6b31e/test/src/test/java/hudson/model/UsageStatisticsTest.java#L158

https://stackoverflow.com/questions/32995559/reverse-a-comparator-in-java-8 suggests `Comparator.reverseOrder()` might be able to handle:
https://github.com/jenkinsci/jenkins/blob/22aa2e6e766074d11249893e3f35e0b99e20d3d0/core/src/main/java/hudson/model/RunMap.java#L159

Comment (jvz) [reply core/src/main/java/hudson/model/Fingerprint.java]: Figures because this feature was added in Java 8, and you're the main person making Java 8 related updated to Jenkins at the moment. ;)

File: core/src/main/java/hudson/model/Queue.java (hunk 4 above)
Comment (jvz) [reply core/src/main/java/hudson/model/Queue.java]: I don't know of an official style guide, but I tend to make the setting for imports at 100 or something really large. Not a huge deal either way.

File: core/src/main/java/hudson/model/Computer.java (hunk 2 above)
Comment (jvz) [reply core/src/main/java/hudson/model/Computer.java]: I mean you can combine the two lines now that it's not so long. More of a style thing, so if you'd prefer to do that in a separate PR, that's fine, too.

Comment (jsoref) [reply core/src/main/java/hudson/model/Computer.java]: Oh, heh. I'll see how smart GitHub diff is. If it's reasonable then I might amend this, if not, I'll add an extra commit.

Comment (jsoref) [reply core/src/main/java/hudson/model/Computer.java]: I checked, it doesn't do a good job of showing the change, so it'll have to be a new commit.

Comment (jvz) [reply core/src/main/java/hudson/model/Computer.java]: Fine with me. A PR for cleaning up formatting makes sense separate from code refactoring.

File: core/src/main/java/jenkins/model/Jenkins.java [hunk 6]
```diff
@@ -3403,9 +3403,6 @@ public void onAttained(Milestone milestone) {
                     LOGGER.log(lv, s);
                 }
             });
-        } catch (InterruptedException | ReactorException | IOException e) {
```
Comment (varyvol) [inline core/src/main/java/jenkins/model/Jenkins.java]: Why is this removed?

**Review by varyvol — COMMENTED** [review]
Really hard to review such amount of changes. In my opinion you should split this kind of PR into several ones.

File: core/src/main/java/jenkins/model/Jenkins.java (hunk 6 above)
Comment (jsoref) [reply core/src/main/java/jenkins/model/Jenkins.java]: The short answer is "because the computer did it". A longer answer is that those exceptions don't apply there.

Comment (varyvol) [reply core/src/main/java/jenkins/model/Jenkins.java]: I guess it's fine since they are not runtime ones.

**Review by varyvol — APPROVED** [review]

**Comment by batmat** [comment]
Let's consider this ready I think.

**Comment by jsoref** [comment]
@batmat: resolved...

**Comment by batmat** [comment]
Let's wait for a green PR build and we merge.

I just updated the changelog proposal...

**Comment by batmat** [comment]
All :green_heart:, merging.
